app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java

```diff
@@ -58,28 +51,15 @@
     protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         this.setContentView(R.layout.activity_qrexchange);
-        this.serviceLocator = NervousFish.getServiceLocator();
+        serviceLocator = NervousFish.getServiceLocator();
 
-        //TODO: Get the user's generated public key from the database
-        final IKeyGenerator keyGenerator = this.serviceLocator.getKeyGenerator();
-        final KeyPair pair = keyGenerator.generateRSAKeyPair("test");
-        this.publicKey = pair.getPublicKey();
-
-        LOGGER.info("Activity created");
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void onActivityResult(final int requestCode, final int resultCode, final Intent intent) {
-        final IntentResult scanResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
-        if (scanResult == null) {
-            LOGGER.error("No scan result in QR Scanner");
-        } else {
-            final String result = scanResult.getContents();
-            addNewContact(result);
+        try {
+            profile = serviceLocator.getDatabase().getProfiles().get(0);
```

Comment:
Again this. If you press Ctrl-Alt-S -> Editor -> Inspections -> Manage -> Import and you import this file: https://www.dropbox.com/s/wryiggq2x9yr0tz/Project_Default.xml?dl=0 , then Intellij will give you a warning about it. If you click "Cleanup code", Intellij will automatically fix all warnings.